### PR TITLE
Enabling links for containers only on bridge network [full ci]

### LINKED ISSUE #3439: Deploying a multi-container wordpress application on VCH via Admiral fails
**VIC version:** 

v0.8.0-rc2-7257-9553959

**Deployment details:**

What was the vic-machine create command used to deploy the VCH?
```
./vic-machine-linux create --target host/Cluster --image-store vsanDatastore --name bhu-vic-host-3 --user Administrator@vsphere.local --password 'password' --no-tlsverify --thumbprint '5B:C9:9B:D8:C3:D7:73:83:00:21:0A:06:B9:F8:CB:61:2A:C6:3A:9B' --public-network 'NSXScalePerf-VMPG01' --volume-store=vsanDatastore/vic-volume:default --insecure-registry 10.197.9.235:80 --force

[[34mINFO[[0m[2016-12-02T21:39:58Z] ### Installing VCH ####
[[33mWARN[[0m[2016-12-02T21:39:58Z] Using administrative user for VCH operation - use --ops-user to improve security (see -x for advanced help)
[[34mINFO[[0m[2016-12-02T21:39:58Z] Loaded server certificate bhu-vic-host-3/server-cert.pem
[[33mWARN[[0m[2016-12-02T21:39:58Z] Configuring without TLS verify - certificate-based authentication disabled
[[34mINFO[[0m[2016-12-02T21:39:59Z] Validating supplied configuration
[[34mINFO[[0m[2016-12-02T21:40:00Z] vDS configuration OK on "bhu-vic-host-3"
[[34mINFO[[0m[2016-12-02T21:40:00Z] Firewall status: DISABLED on "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1801.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:00Z] Firewall status: DISABLED on "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1802.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z] Firewall status: DISABLED on "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1803.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z] Firewall status: DISABLED on "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1804.eng.vmware.com"
[[33mWARN[[0m[2016-12-02T21:40:01Z] Firewall configuration will be incorrect if firewall is reenabled on hosts:
[[33mWARN[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1801.eng.vmware.com"
[[33mWARN[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1802.eng.vmware.com"
[[33mWARN[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1803.eng.vmware.com"
[[33mWARN[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1804.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z] Firewall must permit dst 2377/tcp outbound to VCH management interface if firewall is reenabled
[[34mINFO[[0m[2016-12-02T21:40:01Z] License check OK on hosts:
[[34mINFO[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1801.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1802.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1803.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/w2-hs2-e1804.eng.vmware.com"
[[34mINFO[[0m[2016-12-02T21:40:01Z] DRS check OK on:
[[34mINFO[[0m[2016-12-02T21:40:01Z]   "/VRANSX-PREPROD/host/NSX-SCALE-PERF/Resources"
[[34mINFO[[0m[2016-12-02T21:40:02Z]
[[34mINFO[[0m[2016-12-02T21:40:04Z] Creating virtual app "bhu-vic-host-3"
[[34mINFO[[0m[2016-12-02T21:40:04Z] Creating directory [vsanDatastore] vic-volume
[[34mINFO[[0m[2016-12-02T21:40:04Z] Datastore path is [vsanDatastore] vic-volume
[[34mINFO[[0m[2016-12-02T21:40:04Z] Creating appliance on target
[[34mINFO[[0m[2016-12-02T21:40:04Z] Network role "client" is sharing NIC with "public"
[[34mINFO[[0m[2016-12-02T21:40:04Z] Network role "management" is sharing NIC with "public"
[[34mINFO[[0m[2016-12-02T21:40:09Z] Uploading images for container
[[34mINFO[[0m[2016-12-02T21:40:09Z]   "bootstrap.iso"
[[34mINFO[[0m[2016-12-02T21:40:09Z]   "appliance.iso"
[[34mINFO[[0m[2016-12-02T21:41:47Z] Waiting for IP information
[[34mINFO[[0m[2016-12-02T21:41:59Z] Waiting for major appliance components to launch
[[34mINFO[[0m[2016-12-02T21:41:59Z] Checking VCH connectivity with vSphere target
[[33mWARN[[0m[2016-12-02T21:42:00Z] Could not run VCH vSphere API target check due to ServerFaultCode: The operation is not allowed in the current state.
[[34mINFO[[0m[2016-12-02T21:42:02Z] Initialization of appliance successful
[[34mINFO[[0m[2016-12-02T21:42:02Z]
[[34mINFO[[0m[2016-12-02T21:42:02Z] VCH Admin Portal:
[[34mINFO[[0m[2016-12-02T21:42:02Z] https://10.197.9.234:2378
[[34mINFO[[0m[2016-12-02T21:42:02Z]
[[34mINFO[[0m[2016-12-02T21:42:02Z] Published ports can be reached at:
[[34mINFO[[0m[2016-12-02T21:42:02Z] 10.197.9.234

```

**Steps to reproduce:**
1. Deploy the VCH using the above mentioned command.
2. Register the host on admiral.
3. Upload an application template with the following specification:
```
---
version: "2"
services:
  wordpress:
    image: "wordpress:4.3.1"
    environment:
    - "WORDPRESS_DB_PASSWORD=pass@word01"
    - "WORDPRESS_DB_NAME=wordpress"
    - "WORDPRESS_DB_HOST=mysql:3306"
    ports:
    - "80"
    links:
    - "mysql"
    depends_on:
    - "mysql"
  mysql:
    image: "mariadb:10.0.26"
    environment:
    - "MYSQL_ROOT_PASSWORD=pass@word01"
    ports:
    - "3306"
```
4. Deploy the application.

**Actual behavior:**
The mysql container is sucessfully deployed however, the wordpress container goes offline after reporting the following error in the console:
wordpress-error

**Expected behavior:**
The application should have successfully deployed.

**Logs:**
[container-logs.zip](https://github.com/vmware/vic/files/628547/container-logs.zip)

* If you can access the VCH Admin portal, please download and attach the log bundle(s). See https://vmware.github.io/vic/assets/files/html/vic_admin/log_bundles.html for details.
* If the VCH Admin portal is inaccessible, you can enable SSH to the VCH endpoint VM to obtain logs manually. See https://vmware.github.io/vic/assets/files/html/vic_admin/vch_ssh_access.html for details. The VCH logs will be under /var/log/vic/ on the VM.

**Additional details as necessary:**

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #3439

### CODE REVIEW

File: lib/apiservers/engine/backends/container_proxy.go [hunk 1]
```diff
@@ -888,6 +888,13 @@ func toModelsNetworkConfig(cc types.ContainerCreateConfig) *models.NetworkConfig
 	nc := &models.NetworkConfig{
 		NetworkName: cc.HostConfig.NetworkMode.NetworkName(),
 	}
+
+	// Docker copies Links to NetworkConfig only if it is a UserDefined network, handle that
+	// https://github.com/docker/docker/blame/master/runconfig/opts/parse.go#L598
+	if !cc.HostConfig.NetworkMode.IsUserDefined() && len(cc.HostConfig.Links) > 0 {
+		nc.Aliases = append(nc.Aliases, cc.HostConfig.Links...)
```
Comment (sflxn) [inline lib/apiservers/engine/backends/container_proxy.go]: Is it possible for this line and line 908 to ever happen in one call?

Comment (hmahmood) [inline lib/apiservers/engine/backends/container_proxy.go]: I don't think so, but I fixed the overwriting of nc.Aliases anyway.

**Comment by sflxn** [comment]
lgtm

**Comment by cgtexmex** [comment]
lgtm

**Pull request merged**